### PR TITLE
Improve user profile code quality

### PR DESCRIPTION
**Pull request by peppy**
Second attempt at #5138. This is not a full clean-up effort, just some minor refactors and improvements.

### CODE REVIEW

**Comment by EVAST9919** [comment]
I'd like to remake https://github.com/ppy/osu/pull/5238 after this being merged. It's almost impossible to maintain my current one: you need to hold too many things in your head. And the code itself is hard to follow

**Comment by EVAST9919** [comment]
@peppy need your confirmation

**Comment by smoogipoo** [comment]
Seems to be prone to failure if user is switched too quickly: https://streamable.com/pcmu1
Also there's a weird 1-frame delay when showing a new user: https://streamable.com/2a2n5

I'm not sure if you intended to change the animation, but afaik it detracts from your goals of these types of overlays - having the skeletons present until the user is available.

**Comment by peppy** [comment]
Going to close this one for now. Easier to reapproach it from scratch.
